src/otr/otr.h

```diff
@@ -0,0 +1,170 @@
+ * Franklin Street, Fifth Floor, Boston, MA 02110-1301,USA
+ */
+
+#ifndef IRSSI_OTR_OTR_H
+#define IRSSI_OTR_OTR_H
+
+/* Libotr */
+#include <libotr/proto.h>
+#include <libotr/message.h>
+#include <libotr/context.h>
+#include <libotr/privkey.h>
+
+#include "common.h"
+#include "servers.h"
+
+/* irssi module name */
+#define MODULE_NAME                   "otr/core"
+
+/*
+ * XXX: Maybe this should be configurable?
+ */
+#define OTR_MAX_MSG_SIZE              400
+
+/* OTR protocol id */
+#define OTR_PROTOCOL_ID               "IRC"
+
+#define OTR_DIR                       "otr"
+#define OTR_KEYFILE                   OTR_DIR "/otr.key"
+#define OTR_FINGERPRINTS_FILE         OTR_DIR "/otr.fp"
+#define OTR_INSTAG_FILE               OTR_DIR "/otr.instag"
+
+/*
+ * Specified in OTR protocol version 3. See:
+ * http://www.cypherpunks.ca/otr/Protocol-v3-4.0.0.html
+ */
+#define OTR_MSG_BEGIN_TAG             "?OTR:"
+#define OTR_MSG_END_TAG               '.'
+
+/* IRC /me command marker and len. */
+#define OTR_IRC_MARKER_ME             "/me "
+#define OTR_IRC_MARKER_ME_LEN         sizeof(OTR_IRC_MARKER_ME) - 1
+
+/* Irssi otr user state */
+struct otr_user_state {
+	OtrlUserState otr_state;
+};
+
+/*
+ * Peer OTR internal context.
+ */
+struct otr_peer_context {
+	/* The SMP event status. Used for the Irssi status bar. */
+	OtrlSMPEvent smp_event;
+	/* Did the SMP secret was asked so are we in a responder state? */
+	unsigned int ask_secret;
+	/*
+	 * The fingerprint of the private message OTR session. This is useful for
+	 * the forget command for which we can recover the fingerprint
+	 * automatically.
+	 */
+	Fingerprint *active_fingerprint;
+	/*
+	 * If needed, used to reconstruct the full message from fragmentation.
+	 * Bitlbee for instance does that where we receive a *long* OTR message
+	 * split in multiple PRIVMSG so we need to reconstruct it.
+	 */
+	char *full_msg;
+	/* Size of full_msg. Note this is the allocated memory size. */
+	size_t msg_size;
+	/* Len of the actual string in full_msg NOT counting the NULL byte. */
+	size_t msg_len;
+};
+
+/* given to otr_status_change */
+enum otr_status_event {
+	OTR_STATUS_FINISHED,
+	OTR_STATUS_TRUST_MANUAL,
+	OTR_STATUS_TRUST_SMP,
+	OTR_STATUS_SMP_ABORT,
+	OTR_STATUS_SMP_STARTED,
+	OTR_STATUS_SMP_RESPONDED,
+	OTR_STATUS_SMP_INCOMING,
+	OTR_STATUS_SMP_FINALIZE,
+	OTR_STATUS_SMP_ABORTED,
+	OTR_STATUS_PEER_FINISHED,
+	OTR_STATUS_SMP_FAILED,
+	OTR_STATUS_SMP_SUCCESS,
+	OTR_STATUS_GONE_SECURE,
+	OTR_STATUS_GONE_INSECURE,
+	OTR_STATUS_CTX_UPDATE
+};
+
+enum otr_msg_status {
+	OTR_MSG_ORIGINAL		= 1,
+	OTR_MSG_WAIT_MORE		= 2,
+	OTR_MSG_USE_QUEUE		= 3,
+	OTR_MSG_ERROR			= 4,
+};
+
+/* there can be only one */
+extern struct otr_user_state *user_state_global;
+
+/* Libotr ops functions */
+extern OtrlMessageAppOps otr_ops;
+
+int otr_debug_get(void);
+void otr_debug_toggle(void);
+
+void irssi_send_message(SERVER_REC *irssi, const char *recipient,
```

Comment:
Heh the comment here got marked as outdated because of the debug changes above. Anyway. `irssi_send_message` is questionable naming too

Reply:
Fixed in 2c7da884f9819af910b01ed5aae1ef3f341cbe79